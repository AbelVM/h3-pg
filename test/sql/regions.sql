-- basecell around middle
\set idx_basecell '\'8059fffffffffff\''
-- 7 child hexes in basecell
\set idz_7hex 'ARRAY(SELECT h3_h3_to_children(:idx_basecell, 1))'
-- 6 child hexes in outer rim of basecell
\set idz_6hex 'ARRAY(SELECT id FROM (SELECT h3_h3_to_children(:idx_basecell, 1) id) q WHERE id != \'81583ffffffffff\')'

SELECT h3_h3_get_resolution(:idx_basecell);

-- BACK AND FORTH
SELECT :idz_7hex;
SELECT (h3_h3_set_to_linked_geo(:idz_7hex)).exterior, (h3_h3_set_to_linked_geo(:idz_7hex)).holes;
SELECT h3_polyfill(exterior, holes, 1) FROM 
(
 SELECT *  FROM h3_h3_set_to_linked_geo(:idz_7hex)
) q;

SELECT :idz_6hex;
SELECT (h3_h3_set_to_linked_geo(:idz_6hex)).exterior, (h3_h3_set_to_linked_geo(:idz_6hex)).holes;
SELECT h3_polyfill(exterior, holes, 1) FROM 
(
 SELECT *  FROM h3_h3_set_to_linked_geo(:idz_6hex)
) q;




