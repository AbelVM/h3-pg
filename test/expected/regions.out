-- basecell around middle
\set idx_basecell '\'8059fffffffffff\''
-- 7 child hexes in basecell
\set idz_7hex 'ARRAY(SELECT h3_h3_to_children(:idx_basecell, 1))'
-- 6 child hexes in outer rim of basecell
\set idz_6hex 'ARRAY(SELECT id FROM (SELECT h3_h3_to_children(:idx_basecell, 1) id) q WHERE id != \'81583ffffffffff\')'
SELECT h3_h3_get_resolution(:idx_basecell);
 h3_h3_get_resolution 
----------------------
                    0
(1 row)

-- BACK AND FORTH
SELECT :idz_7hex;
                                                       array                                                       
-------------------------------------------------------------------------------------------------------------------
 {81583ffffffffff,81587ffffffffff,8158bffffffffff,8158fffffffffff,81593ffffffffff,81597ffffffffff,8159bffffffffff}
(1 row)

SELECT (h3_h3_set_to_linked_geo(:idz_7hex)).exterior, (h3_h3_set_to_linked_geo(:idz_7hex)).holes;
                                                                                                                                                                                                                                                                                                                                                                         exterior                                                                                                                                                                                                                                                                                                                                                                          | holes 
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------
 ((13.9151838856346,23.4504377153203),(9.38067660520161,25.6037025769688),(5.36757370512317,23.0111470503629),(0.994677192549833,24.8583358344496),(-2.48386501194795,22.1975413863024),(-1.71130990397791,17.9354045818174),(-3.29256482968365,16.7189804058919),(-4.86687202658957,15.2198852193985),(-4.01399844347047,11.5452959754148),(-0.319182726856614,9.73183667097592),(0.309803963535533,5.86451836551763),(3.94303615578646,3.96879697660959),(7.21291456272007,5.86039216837659),(9.10237141854828,4.86081181207326),(10.7227778272856,3.78962499175013),(14.5832500714117,5.76286049143693),(14.4350416743716,9.91339077600533),(18.371312883447,12.0234431609871),(18.3651793828161,16.5059476035611),(14.1025321744588,18.8007437662402)) | {}
(1 row)

SELECT h3_polyfill(exterior, holes, 1) FROM 
(
 SELECT *  FROM h3_h3_set_to_linked_geo(:idz_7hex)
) q;
   h3_polyfill   
-----------------
 8158fffffffffff
 8159bffffffffff
 8158bffffffffff
 81597ffffffffff
 81587ffffffffff
 81593ffffffffff
 81583ffffffffff
(7 rows)

SELECT :idz_6hex;
                                               array                                               
---------------------------------------------------------------------------------------------------
 {81587ffffffffff,8158bffffffffff,8158fffffffffff,81593ffffffffff,81597ffffffffff,8159bffffffffff}
(1 row)

SELECT (h3_h3_set_to_linked_geo(:idz_6hex)).exterior, (h3_h3_set_to_linked_geo(:idz_6hex)).holes;
                                                                                                                                                                                                                                                                                                                                                                         exterior                                                                                                                                                                                                                                                                                                                                                                          |                                                                                                            holes                                                                                                             
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ((18.371312883447,12.0234431609871),(18.3651793828161,16.5059476035611),(14.1025321744588,18.8007437662402),(13.9151838856346,23.4504377153203),(9.38067660520161,25.6037025769688),(5.36757370512317,23.0111470503629),(0.994677192549833,24.8583358344496),(-2.48386501194795,22.1975413863024),(-1.71130990397791,17.9354045818174),(-3.29256482968365,16.7189804058919),(-4.86687202658957,15.2198852193985),(-4.01399844347047,11.5452959754148),(-0.319182726856614,9.73183667097592),(0.309803963535533,5.86451836551763),(3.94303615578646,3.96879697660959),(7.21291456272007,5.86039216837659),(9.10237141854828,4.86081181207326),(10.7227778272856,3.78962499175013),(14.5832500714117,5.76286049143693),(14.4350416743716,9.91339077600533)) | {"((2.30284280920586,16.0986886429595),(5.88226299580354,18.5160524000961),(10.0979776613705,16.4491832881973),(10.4162881227623,12.0573252262095),(6.80116931700978,9.89974307082342),(2.88810760150022,11.881770859939))"}
(1 row)

SELECT h3_polyfill(exterior, holes, 1) FROM 
(
 SELECT *  FROM h3_h3_set_to_linked_geo(:idz_6hex)
) q;
   h3_polyfill   
-----------------
 8158fffffffffff
 8159bffffffffff
 8158bffffffffff
 81597ffffffffff
 81587ffffffffff
 81593ffffffffff
(6 rows)

